### PR TITLE
Add support for JsonWireProtocol element/:id/location_in_view

### PR DESCRIPTION
**Pull request by kaleposhobios**
This method scrolls the element into view then 
returns its location.

Reference: https://github.com/facebook/WebDriverAgent/issues/89

### CODE REVIEW

File: XCTWebDriverAgentLib/Commands/FBElementCommands.m [hunk 1]
```diff
@@ -58,6 +58,17 @@ + (NSArray *)routes
         XCUIElement *element = [elementCache elementForIndex:[request.parameters[@"id"] integerValue]];
         return FBResponseDictionaryWithStatus(FBCommandStatusNoError, element.wdLocation);
     }],
+    [[FBRoute GET:@"/element/:id/location_in_view"] respond:^ id<FBResponsePayload> (FBRouteRequest *request) {
+        FBXCTElementCache *elementCache = (FBXCTElementCache *)request.session.elementCache;
+        XCUIElement *element = [elementCache elementForIndex:[request.parameters[@"id"] integerValue]];
```
Comment (marekcirkos) [inline XCTWebDriverAgentLib/Commands/FBElementCommands.m]: Could you refactor code a bit and reuse `+ (id<FBResponsePayload>)handleScrollElementToVisible: withRequest:` instead?
Eg. You could add method that returns `BOOL` and use it inside `+ (id<FBResponsePayload>)handleScrollElementToVisible: withRequest:` an in your method.

**Comment by kaleposhobios** [comment]
Thanks for the feedback. The code is now refactored and properly DRY.

File: XCTWebDriverAgentLib/Commands/FBElementCommands.m [hunk 2]
```diff
@@ -239,10 +248,31 @@ + (NSArray *)routes
   [element resolve];
   [element scrollToVisible];
   [element resolve];
-  if (!element.isFBVisible) {
```
Comment (marekcirkos) [inline XCTWebDriverAgentLib/Commands/FBElementCommands.m]: I don't think those are needed if you call `scrollElementToVisible` :)

**Comment by kaleposhobios** [comment]
@marekcirkos Oops. Thanks -- now fixed.

**Comment by marekcirkos** [comment]
Awesome! Thank you!

**Comment by marekcirkos** [comment]
I needed to revert your diff due to issues with our bot. They have been merged back here fa239bd

**Pull request merged**